src/org/org.ts

```diff
@@ -780,42 +784,41 @@ export class Org extends AsyncOptionalCreatable<Org.Options> {
    * @private
    */
   private async deleteSandbox(prodOrg?: Org): Promise<void> {
+    const sandbox = await this.getSandboxConfig(this.getOrgId());
     prodOrg ??= await Org.create({
       aggregator: this.configAggregator,
-      aliasOrUsername: await this.getSandboxOrgConfigField(SandboxOrgConfig.Fields.PROD_ORG_USERNAME),
+      aliasOrUsername: sandbox?.prodOrgUsername,
     });
-    let result: { SandboxInfoId: string };
-
-    // attempt to locate sandbox id by username
-    try {
-      // try to calculate sandbox name from the production org
-      // production org: admin@integrationtesthub.org
-      // this.getUsername: admin@integrationtesthub.org.dev1
-      // sandboxName in Production: dev1
-      const sandboxName = (this.getUsername() || '').split(`${prodOrg.getUsername()}.`)[1];
-      if (!sandboxName) {
-        this.logger.debug('Could not construct a sandbox name');
-        throw new Error();
-      }
-      this.logger.debug(`attempting to locate sandbox with username ${sandboxName}`);
-      result = await this.queryProduction(prodOrg, 'SandboxName', sandboxName);
-      if (!result) {
-        this.logger.debug(`Failed to find sandbox with username: ${sandboxName}`);
-        throw new Error();
-      }
-    } catch {
-      // if an error is thrown, don't panic yet. we'll try querying by orgId
-      const trimmedId = sfdc.trimTo15(this.getOrgId()) as string;
-      this.logger.debug(`defaulting to trimming id from ${this.getOrgId()} to ${trimmedId}`);
+    let sandboxInfoId: string | undefined = sandbox?.sandboxInfoId;
+    if (!sandboxInfoId) {
```

Comment:
If not available, follow steps to fetch sandboxInfoId by querying sandbox process with sandbox name and/or sandbox org id